src/sysusers/sysusers.c

```diff
@@ -1785,27 +1861,33 @@ int main(int argc, char *argv[]) {
 
         if (optind < argc) {
                 int j;
-
-                for (j = optind; j < argc; j++) {
-                        k = read_config_file(argv[j], false);
-                        if (k < 0 && r == 0)
-                                r = k;
+                bool found_override = false;
+
+                if (arg_config_name) {
+                        /* If overrides should be allowed, look for a config file that
+                         * would override our configuration: treat the configuration
+                         * given on the command as if it was in priority between /run
+                         * and /usr/. */
+                        r = read_config_files(override_file_dirs, arg_config_name);
+                        if (r < 0)
+                                goto finish;
+                        found_override = r;
                 }
-        } else {
-                _cleanup_strv_free_ char **files = NULL;
-                char **f;
 
-                r = conf_files_list_nulstr(&files, ".conf", arg_root, 0, conf_file_dirs);
-                if (r < 0) {
-                        log_error_errno(r, "Failed to enumerate sysusers.d files: %m");
+                if (!found_override)
+                        for (j = optind; j < argc; j++) {
+                                if (arg_direct)
+                                        /* Use (argument):n, where n==1 for the first positional arg */
+                                        k = parse_line("(argument)", j - optind + 1, argv[j]);
+                                else
+                                        k = read_config_file(argv[j], false);
+                                if (k < 0 && r == 0)
+                                        r = k;
```

Comment:
The value of `r` never used. I think it should be checked after the loop, right?